README.md

```diff
@@ -69,46 +69,146 @@ fields that can be selected, so implementations must provide this out-of-band. I
+Item should be included.
+4. If `exclude` is specified and `include` is null or an empty
+array, then the `exclude` fields should be excluded from the default set.
+5. For nested fields (e.g., `properties.datetime`), the most specific path
+should be honored first, and `include` should be preferred over `exclude`. For
+example:
+    1. If a field is in `exclude`, and a nested field of that field is in
+    `include`, the nested field should be included, but no other nested
+    fields in the field should be included.  For example, if `properties` is
+    excluded and `properties.datetime` is included, then `datetime`
+    should be the only nested field in `properties`.
+    2. If a field is in `include`, and a nested field of that field is in `exclude`, the field
+    should be included, and the nested field should be excluded.  For example,
+    if `properties` is included and `properties.datetime` is excluded, then
+    `datetime` should not be in `properties`.
+6. If the same field is present in both `include` and `exclude`, it should be included.
+
+### `null` vs. empty vs. missing
+
+There is a semantic difference between a missing value (i.e., if `include` is not
+in the JSON object), and a `null` or empty value.  The recommended behavior
+around missing vs. null and empty values is described in [the section
+above](#includeexclude-semantics), but is summarized here for reference. "ALL"
+means that all of the Item's fields should be returned; "DEFAULT" means the
+default set (i.e. the required fields for a valid STAC Item) should be returned.
+
+| include | exclude | returned |
+| -- | -- | -- |
+| missing, `null`, or empty | missing, `null`, or empty | DEFAULT |
+| ["a", "b"] | missing, `null`, or empty | ["a", "b"] |
+| missing | ["a", "b"] | ALL except for ["a", "b"] |
+| `null` or empty | ["a", "b"] | DEFAULT except for ["a", "b"] |
+| ["a", "b"] | ["a"] | ["a", "b"] |
+| ["a.b"] | ["a"] | `a.b` should be the only field in `a` |
+| ["a"] | ["a.b"] | `a` should be included, but it should not include `a.b` |
+
+In this example, `include` is missing:
+
+```json
+{
+  "fields": {
+    "exclude": ["geometry"]
+  }
+}
+```
+
+In these two examples, `include` is `null` and empty, respectively:
+
+```json
+{
+  "fields": {
+    "include": null,
+    "exclude": ["geometry"]
+  }
+}
+```
+
+```json
+{
+  "fields": {
+    "include": [],
+    "exclude": ["geometry"]
+  }
+}
+```
+
+The special case of both `include` and `exclude` missing is possible both in
+JSON and text. In JSON:
+
+```json
+{
+  "fields": {}
+}
+```
+
+or
+
+```json
+{
+  "fields": null
+}
+```
+
+In text:
+
+```text
+?fields=
+```
+
+It is not possible to differentiate between missing, null, and empty for the
+text representation, so implementations should assume the value is empty, NOT
+missing. For example, this is a case of `include` being empty (NOT missing):
+
+```text
+?fields=-geometry
+```
 
 ## Examples
```

Comment:
these need subheadings -- i had a hard time figuring out if the descriptions went with the code blocks above or below when dropped in the middle.